api/v1beta1/azurecluster_validation.go

```diff
@@ -252,6 +259,11 @@ func validateSubnets(subnets Subnets, vnet VnetSpec, fldPath *field.Path) field.
 			allErrs = append(allErrs, validatePrivateEndpoints(subnet.PrivateEndpoints, subnet.CIDRBlocks, fldPath.Index(i).Child("privateEndpoints"))...)
 		}
 	}
+
+	if clusterSubnet {
```

Comment:
I'm a bit confused about what `clusterSubnet` means. Does this mean the subnet belongs to a cluster, and there are no required subnet roles?

Reply:
yes, it can be used as a common subnet for both node and controlplane subnet.

Reply:
can we add a few code comments in this validation function to explain that logic?